File: Java/Exercicio 1.java
// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int N) {
        //variables
        String binary;
        String charBinary;
        int maxsequencial=0;
        int sequencial=0;

        //convert int to binary
        binary = Integer.toBinaryString(N);
        //System.out.println("Number:" + N + ", binary:" + binary);
        
        for(int i=0; i< binary.length(); i++){
            if(binary.charAt(i) == '0'){
                sequencial +=1;        
            }
            else{
                if(maxsequencial<sequencial){
                    maxsequencial = sequencial;
                }   
                sequencial = 0;
            }
            
        }
        return maxsequencial;
    }
}

File: Visual Basic/Exercicio 1.vb
'Find a maximal sequence of consecutive zeros in a binary gap

Private Function solution(N As Integer) As Integer
    'variables
    Dim  binary as String
    Dim charBinary as String
    Dim maxsequencial as Integer = 0
    Dim sequencial as Integer = 0
    
    'convert int to binary
    binary = Convert.ToString(N, 2)
    'Console.WriteLine("Number:{0}, binary:{1}",N, binary)
    
    For i As Integer = 0 To binary.Length -1 
        charBinary = binary.Substring(i,1)
        if charBinary = "0" then
            sequencial +=1
        else
            if maxsequencial<sequencial then
                maxsequencial = sequencial
            end if   
            sequencial = 0
        end if
    next i
    return maxsequencial
    
    return 0
End Function